### DISCUSSION TITLE
NewType errors to help with using NewType

### DISCUSSION THREAD

**Discussion by rchiodo** [microsoft/pylance-release#3511] [Enhancement]
When using NewType, it would be nice to give better error messages and better representation:

![image](https://user-images.githubusercontent.com/19672699/196812827-c26d40f8-308a-412e-be14-c2ec1a5db1b7.png)

NewTypes aren't referenced that way anywhere else. It would be cool if that said e: NewType[int]

When creating NewTypes, you can only pass the base object as a parameter. Error messages on incorrect usage could indicate this:

![image](https://user-images.githubusercontent.com/19672699/196813021-43231e1e-4948-49f8-87dc-02e72b77ecb8.png)

**Comment by erictraut** [comment]
`NewType` is not a type itself. It's a function that creates a new class with a user-specified name. This new class is treated by a static type checker as a subclass of a specified type, and it has a constructor that accepts only an instance of the original type.

In your example above, you chose the name `e` for this type, so an instance of this class has the type `e`, and the class itself is type `type[e]`. This is consistent with other object and class types. To display it otherwise would introduce a confusing inconsistency, in my opinion.

Typically, a more descriptive class name would be chosen by the user. For example:

```python
Width = NewType('Width', int)

x = Width(3)
reveal_type(x) # Width
```
